BuildingBlocks/src/UnitTestTools/FluentValidation/Extensions.cs

```diff
@@ -31,3 +33,15 @@ public static class IValidatorExtensions
         return validationResult.Errors;
     }
 }
+
+public static class ValidationTestExtensions
+{
+    public static void ShouldHaveValidationErrorForItem<T>(this TestValidationResult<T> testValidationResult, string propertyName, string expectedErrorCode,
+        string expectedErrorMessage)
+    {
+        var validationError = testValidationResult.Errors.FirstOrDefault(r => r.PropertyName.Contains(propertyName)) ?? throw new XunitException($"Expected error for property '{propertyName}'.");
```

Comment:
I don't think that `FirstOrDefault` is a good idea. If there is more than one error for the property, the assertion might fail even though there IS an error that matches the expected criteria

Reply:
Why not

```c#
errorsForProperty.Should().Contain(r => r.ErrorCode == expectedErrorCode && r.ErrorMessage == expectedErrorMessage);
```
?